src/ansi-c/c_typecheck_base.cpp

```diff
@@ -717,7 +722,11 @@ void c_typecheck_baset::typecheck_declaration(
       else
       {
         std::string asm_name;
-        asm_name=id2string(full_spec.section)+"$$";
+        asm_name = id2string(full_spec.section);
+        std::string::size_type primary_section = asm_name.find('.', 1);
```

Comment:
Worth an `INVARIANT(asm_name.size() >= 1 && asm_name[0] == '.')` to check you have the format you think you do?